vividus-plugin-azure-resource-manager/src/main/java/org/vividus/azure/client/AzureResourceManagerClient.java

```diff
@@ -0,0 +1,144 @@
+     *                                vaults/sample-vault</code>.
+     * @param apiVersion              Azure resource provider API version. Note API versions may vary depending on
+     *                                the resource type.
+     * @param azureResourceBody       The Azure resource configuration in JSON format.
+     * @param responseBodyConsumer    The Consumer that handles the response body if the request is successful.
+     */
+    public void executeHttpRequest(HttpMethod method, String azureResourceIdentifier, String apiVersion,
+                                   Optional<String> azureResourceBody, Consumer<String> responseBodyConsumer)
+    {
+        String url = String.format("%ssubscriptions/%s%s?api-version=%s",
+                azureProfile.getEnvironment().getResourceManagerEndpoint(), azureProfile.getSubscriptionId(),
+                StringUtils.prependIfMissing(azureResourceIdentifier, "/"), apiVersion);
+        executeHttpRequest(method, url, azureResourceBody, responseBodyConsumer, softAssert::recordFailedAssertion);
+    }
+
+    /**
+     * Executes an HTTP request to the specified URL using the provided parameters.
+     * This method allows for flexible error handling based on the availability of a soft assertion mechanism.
+     * If softAssert is available, it uses soft assertions to handle errors; otherwise,
+     * it throws RuntimeException on errors.
+     *
+     * @param method               The HTTP method to be used for the request (e.g., GET, POST, PUT, DELETE).
+     * @param azureResourceUrl     It's used to specify Azure resource uniquely. For example:
+     *                             <ul>
+     *                             <li><code>https://management.azure.com/subscriptions/
+     *                             00000000-0000-0000-0000-000000000000/resourceGroups/sample-resource-group/
+     *                             providers/Microsoft.KeyVault/vaults/sample-vault?api-version=2021-10-01</code>
+     *                             <br><i>or</i>
+     *                             <br>
+     *                             <li><code>https://api.loganalytics.io/v1/workspaces/00000000-0000-0000-0000-000000000000/
+     *                             query?query=Syslog</code>
+     *                             </ul>
+     * @param azureResourceBody    The Azure resource configuration in JSON format.
+     * @param responseBodyConsumer The Consumer that handles the response body if the request is successful.
+     */
+    public void executeHttpRequest(HttpMethod method, String azureResourceUrl, Optional<String> azureResourceBody,
+                                    Consumer<String> responseBodyConsumer)
+    {
+        Consumer<String> errorCallback = error -> { throw new RuntimeException(error); };
+        if (softAssert != null)
+        {
+            errorCallback = softAssert::recordFailedAssertion;
+        }
```

Comment:
this logic should be provided by code invoking this method

Reply:
moved into ResourceManagementSteps